src/hotspot/cpu/aarch64/aarch64_vector_ad.m4

```diff
@@ -3616,6 +3616,47 @@ instruct vmaskcmp_sve(pReg dst, vReg src1, vReg src2, immI cond, rFlagsReg cr) %
   %}
   ins_pipe(pipe_slow);
 %}
+dnl
+dnl VMASKCMP_SVE_IMM_I($1      , $2            )
+dnl VMASKCMP_SVE_IMM_I(type_imm, type_condition)
+define(`VMASKCMP_SVE_IMM_I', `
+instruct vmask$2_immI_sve(pReg dst, vReg src, $1 imm, immI_$2_cond cond, rFlagsReg cr) %{
+  predicate(UseSVE > 0);
+  match(Set dst (VectorMaskCmp (Binary src (ReplicateB imm)) cond));
+  match(Set dst (VectorMaskCmp (Binary src (ReplicateS imm)) cond));
+  match(Set dst (VectorMaskCmp (Binary src (ReplicateI imm)) cond));
+  effect(KILL cr);
+  format %{ "vmask$2_immI_sve $dst, $src, $imm, $cond\t# KILL cr" %}
+  ins_encode %{
+    Assembler::Condition condition = to_assembler_cond((BoolTest::mask)$cond$$constant);
+    BasicType bt = Matcher::vector_element_basic_type(this);
+    uint length_in_bytes = Matcher::vector_length_in_bytes(this);
+    assert(length_in_bytes == MaxVectorSize, "invalid vector length");
+    __ sve_cmp(condition, $dst$$PRegister, __ elemType_to_regVariant(bt),
+               ptrue, $src$$FloatRegister, (int)$imm$$constant);
+  %}
+  ins_pipe(pipe_slow);
+%}')dnl
+dnl VMASKCMP_SVE_IMM_L($1      , $2            )
+dnl VMASKCMP_SVE_IMM_L(type_imm, type_condition)
+define(`VMASKCMP_SVE_IMM_L', `
+instruct vmask$2_immL_sve(pReg dst, vReg src, $1 imm, immI_$2_cond cond, rFlagsReg cr) %{
+  predicate(UseSVE > 0);
+  match(Set dst (VectorMaskCmp (Binary src (ReplicateL imm)) cond));
+  effect(KILL cr);
+  format %{ "vmask$2_immL_sve $dst, $src, $imm, $cond\t# KILL cr" %}
+  ins_encode %{
+    Assembler::Condition condition = to_assembler_cond((BoolTest::mask)$cond$$constant);
+    uint length_in_bytes = Matcher::vector_length_in_bytes(this);
+    assert(length_in_bytes == MaxVectorSize, "invalid vector length");
+    __ sve_cmp(condition, $dst$$PRegister, __ D, ptrue, $src$$FloatRegister, (int)$imm$$constant);
+  %}
+  ins_pipe(pipe_slow);
+%}')dnl
+VMASKCMP_SVE_IMM_I(immI5, cmp)
```

Comment:
This is tricky to review because the two macros here seems to be almost, but not exactly, the same. Why is that?

Reply:
This patch adds rules for vector comparing with immediate. These immediate may have different
types and his manifests as different ConNodes in middle-end (ConI and ConL). ConI and ConL have 
different methods to get the value, i.e., get_int() for ConI and get_long() for ConL. We should use this value in predicate, so I set two macros, one for integer (byte, short and int) and another one for long.

Reply:
Please show me where these differences appear. I can't see them.